### PR TITLE
Refactor initializer

### PR DESCRIPTION
**Pull request by mydea**
After upgrading to the new test syntax, the private functionality the initializer uses breaks.

Especially, this part doesn't work anymore in the new style tests:

```js
 const container = app.__container__; 
  let config; 
  if(container.factoryFor) { 
    config = container.factoryFor('config:environment').class; 
  } else { 
    config = container.lookupFactory('config:environment'); 
  } 
```

This PR first updates all the dependencies to their latest versions, and then refactors the initializer to work with the new syntax.

This should be a completely internal change - to the end user, nothing should change.

It turned out, the hardest part was to get it to work with Fastboot. The old way of fetching that data from a meta tag didn't work there, as Fastboot doesn't have access to the DOM. 
Because of this, I refactored it to instead write the URL/inline form into a dedicated file, and read it from there. 
At the end, this should not change anything for a consumer of this addon.

One thing I figured out, is that if you're using Fastboot, you need to use the inline form, as Fastboot cannot use jQuery to fetch the assetMap. If it detects that you are building for Fastboot, it will automatically force you into `inline: true` mode, and say so on the console (so you know what's going on).  Generally, I think it would be a good idea to recommend inlining it anyhow - it will be put into the `vendor.js` file, and you save one request. Also, I found some issues with assetMap fingerprinting (e.g. https://github.com/rickharrison/broccoli-asset-rev/issues/122, and also some issues with fastboot) - so inlining is definitely the safest option. The other option still works the same as before, though.

### CODE REVIEW

**Comment by RuslanZavacky** [comment]
Ooh, that's an amazing PR @mydea, thank you very much! Will look into it shortly.

File: addon/services/asset-map.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
-import Ember from 'ember';
+import Service from '@ember/service';
+import { computed, get } from '@ember/object';
 
-export default Ember.Service.extend({
+export default Service.extend({
   enabled: false,
-  map: {},
+  map: computed(function() {
```
Comment (RuslanZavacky) [inline addon/services/asset-map.js]: Small nitpick, could be as

```js
map: computed(() => ({})),
```

**Comment by mydea** [comment]
I have added one more commit, fixing an issue I noticed when building for production. I also made the small adaption you requested, and disabled `isDevelopingAddon`.

**Comment by zeppelin** [comment]
@RuslanZavacky Running this in production, works like charm! Care to cut a release on NPM? ;)

**Comment by RuslanZavacky** [comment]
Thank you, @mydea! Thats a great contribution ❤️

**Comment by mydea** [comment]
Awesome! Just noting, not sure if that is on purpose, but the 0.7.0 release is not on NPM yet?

**Comment by RuslanZavacky** [comment]
@mydea published now :)

**Pull request merged**